retriever/lib/scripts.py

```diff
@@ -99,13 +105,224 @@ def SCRIPT_LIST():
+    if matches:
+        print("Did you mean:"
+              " \n\t{}".format("\n\t".join([i[0] for i in matches])))
+
+
 def get_script(dataset):
     """Return the script for a named dataset."""
     scripts = {script.name: script for script in SCRIPT_LIST()}
     if dataset in scripts:
-        return scripts[dataset]
+        script = scripts[dataset]
+        local_version = script.version
+        if dataset in RETRIEVER_DATASETS:
+            upstream_version = get_script_version_upstream(dataset, repo=RETRIEVER_REPOSITORY)
+        else:
+            upstream_version = get_script_version_upstream(dataset)
+        if upstream_version is None or LooseVersion(local_version) >= LooseVersion(upstream_version):
+            return script
+        prompt = "A newer version of {dataset} is available. Would you like to download " \
+                 "it? (y/N): ".format(dataset=dataset)
+        should_download = input(prompt)
+        while not (should_download.lower() in ['y', 'n', '']):
+            print("Please enter either y or n.")
+            should_download = input()
+        if should_download.lower() == 'y':
+            if dataset in RETRIEVER_DATASETS:
+                read_script = get_script_upstream(dataset, repo=RETRIEVER_REPOSITORY)
+            else:
+                read_script = get_script_upstream(dataset)
+            if read_script is None:
+                print("Unable to download {dataset}.".format(dataset=dataset))
+                return script
+            return read_script
+        return script
     else:
-        raise KeyError("No dataset named: {}".format(dataset))
+        if dataset in RETRIEVER_DATASETS:
+            read_script = get_script_upstream(dataset, repo=RETRIEVER_REPOSITORY)
+        else:
+            read_script = get_script_upstream(dataset)
+        if read_script is None:
+            raise KeyError("No dataset named: {}".format(dataset))
+        return read_script
+
+
+def get_script_upstream(dataset, repo=REPOSITORY):
+    """Return the upstream script for a named dataset."""
+    try:
+        is_json = True
+        script = dataset.replace('-', '_')
+        script_name = script + ".json"
+        filepath = "scripts/" + script_name
+        newpath = os.path.normpath(os.path.join(SCRIPT_WRITE_PATH, script_name))
+        r = requests.get(repo + filepath, allow_redirects=True, stream=True)
+        if r.status_code == 404:
+            is_json = False
+            script_name = script + ".py"
+            filepath = "scripts/" + script_name
+            newpath = os.path.normpath(os.path.join(SCRIPT_WRITE_PATH, script_name))
+            r = requests.get(repo + filepath, allow_redirects=True, stream=True)
+            if r.status_code == 404:
+                return None
+        with open(newpath, 'wb') as f:
+            for chunk in r.iter_content(chunk_size=1024):
+                f.write(chunk)
+        r.close()
+        if is_json:
```

Comment:
Remove  `Unnecessary "else" after "return"`

Reply:
We can also improve on the variable names we are using by making them more descriptive instead of single characters, where applicable.